lightbug_http/client.mojo

```diff
@@ -9,29 +9,42 @@ from lightbug_http.libc import (
     close,
 )
 from lightbug_http.strings import to_string
-from lightbug_http.io.bytes import Bytes
-from lightbug_http.utils import ByteReader
-from lightbug_http.net import create_connection, default_buffer_size
+from lightbug_http.net import default_buffer_size
 from lightbug_http.http import HTTPRequest, HTTPResponse, encode
 from lightbug_http.header import Headers, HeaderKey
+from lightbug_http.net import create_connection, SysConnection
+from lightbug_http.io.bytes import Bytes
+from lightbug_http.utils import ByteReader
+from collections import Dict
 
 
 struct Client:
     var host: StringLiteral
     var port: Int
     var name: String
 
+    var _connections: Dict[String, SysConnection]
+
     fn __init__(inout self) raises:
         self.host = "127.0.0.1"
         self.port = 8888
         self.name = "lightbug_http_client"
+        self._connections = Dict[String, SysConnection]()
 
     fn __init__(inout self, host: StringLiteral, port: Int) raises:
         self.host = host
         self.port = port
         self.name = "lightbug_http_client"
+        self._connections = Dict[String, SysConnection]()
 
-    fn do(self, owned req: HTTPRequest) raises -> HTTPResponse:
+    fn __del__(owned self):
+        for conn in self._connections.values():
+            try:
+                conn[].close()
+            except:
+                pass
+
```

Comment:
Should we log or throw an error here if the connection couldn't be closed? Might be an overkill though

Reply:
`__del__` can't raise, and I don't think libraries should generally print things but I do think we should be implementing some optional debug logging throughout the library. I'll add a comment noting that this is a good place to have one once we've implemented that.